pymc3/distributions/discrete.py

```diff
@@ -603,6 +604,107 @@ def logcdf(value, mu):
+            p = omega * c * (1 + lam / c)**(x - 1) * p / x
+            s += p
+        return x
+
+
+generalized_poisson = GeneralizedPoissonRV()
+
+
+class GeneralizedPoisson(Discrete):
+    R"""
+    Generalized Poisson log-likelihood.
+
+    Used to model count data that can be either overdispersed or underdispersed.
+    Offers greater flexibility than the standard Poisson which assumes equidispersion,
+    where the mean is equal to the variance.
+
+    The PMF is taken from
+        Consul, PoC, and Felix Famoye. "Generalized Poisson regression model." 
+        Communications in Statistics-Theory and Methods 21.1 (1992): 89-109.
+
+    The generator function is taken from
+        Famoye, Felix. "Generalized Poisson random variate generation." American
+        Journal of Mathematical and Management Sciences 17.3-4 (1997): 219-237.
+
+    The pmf of this distribution is
+
+    .. math:: f(x \mid \theta, \lambda) = 
+                  \frac{\theta (\theta + \lambda x)^{x-1} e^{-\theta - \lambda x}}{x!}
+
+    ========  ======================================
+    Support   :math:`x \in \mathbb{N}_0`
+    Mean      :math:`\frac{\theta}{1 - \lambda}`
+    Variance  :math:`\frac{\theta}{(1 - \lambda)^3}`
+    ========  ======================================
+
+    Parameters
+    ----------
+    theta: float
+        Mean parameter (theta > 0).
+    lam: float
+        Dispersion parameter (max(-1, -theta/4) <= lam <= 1).
+
+    Notes
+    -----
+    When lam = 0, the Generalized Poisson reduces to the standard Poisson with mu = theta.
+    When lam < 0, the mean is greater than the variance (underdispersion).
+    When lam > 0, the mean is less than the variance (overdispersion).
+    """
+    rv_op = generalized_poisson
+
+    @classmethod
+    def dist(cls, theta, lam, *args, **kwargs):
+        theta = at.as_tensor_variable(floatX(theta))
+        lam = at.as_tensor_variable(floatX(lam))
+        return super().dist([theta, lam], *args, **kwargs)
+
+    def logp(value, theta, lam):
+        r"""
+        Calculate log-probability of Generalized Poisson distribution at specified value.
+
+        Parameters
+        ----------
+        value: numeric
+            Value(s) for which log-probability is calculated. If the log probabilities for multiple
+            values are desired the values must be provided in a numpy array or Aesara tensor
+
+        Returns
+        -------
+        TensorVariable
+        """
+        log_prob = bound(np.log(theta) + logpow(theta + lam * value, value - 1)
+                         - (theta + lam * value) - factln(value),
+                         theta > 0,
+                         -1 <= lam, -theta/4 <= lam, lam <= 1,
+                         value >= 0)
+        # Probability is 0 when value > m, where m is the largest positive integer for which
+        # theta + m * lam > 0 (when lam < 0).
+        return at.switch(theta + value * lam <= 0, -np.inf, log_prob)
```

Comment:
I tried to simplify the logp expression. What do you think? (Please check if I didn't add any mistakes)

```suggestion
        theta_lam_value = theta + lam * value
        log_prob = np.log(theta) + logpow(theta_lam_value, value - 1) - theta_lam_value - factln(value)
        
        # Probability is 0 when value > m, where m is the largest positive integer for which
        # theta + m * lam > 0 (when lam < 0).
        log_prob = at.switch(theta_lam_value <= 0, -np.inf, log_prob)
        
        return bound(
                log_prob,
                value >= 0,
                theta > 0,
                abs(lam) <= 1, 
                -theta/4 <= lam, 
        )
```

Reply:
Looks good to me!